File: openmrs-android-sdk/src/main/java/com/example/openmrs_android_sdk/library/databases/entities/VisitEntity.java
package com.example.openmrs_android_sdk.library.databases.entities;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;

import com.example.openmrs_android_sdk.library.models.Resource;

@Entity(tableName = "visits")
public class VisitEntity extends Resource {
    @NonNull
    @ColumnInfo(name = "patient_id")
    private long patientKeyID;
    @ColumnInfo(name = "visit_type")
    private String visitType;
    @ColumnInfo(name = "visit_place")
    private String visitPlace;
    @NonNull
    @ColumnInfo(name = "start_date")
    private String startDate;
    @ColumnInfo(name = "stop_date")
    private String stopDate;

    public VisitEntity() {
    }

    public void setPatientKeyID(long patientKeyID) {
        this.patientKeyID = patientKeyID;
    }

    public void setVisitType(String visitType) {
        this.visitType = visitType;
    }

    public void setVisitPlace(String visitPlace) {
        this.visitPlace = visitPlace;
    }

    public void setStartDate(String startDate) {
        this.startDate = startDate;
    }

    public void setStopDate(String stopDate) {
        this.stopDate = stopDate;
    }

    public long getPatientKeyID() {
        return patientKeyID;
    }

    public String getVisitType() {
        return visitType;
    }

    public String getVisitPlace() {
        return visitPlace;
    }

    @NonNull
    public String getStartDate() {
        return startDate;
    }

    public String isStartDate() {
        return startDate;
    }

    public String getStopDate() {
        return stopDate;
    }
}

File: openmrs-client/src/main/java/org/openmrs/mobile/databases/entities/VisitEntity.kt
/*
 * The contents of this file are subject to the OpenMRS Public License
 * Version 1.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 * http://license.openmrs.org
 *
 * Software distributed under the License is distributed on an "AS IS"
 * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
 * License for the specific language governing rights and limitations
 * under the License.
 *
 * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
 */

package org.openmrs.mobile.databases.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import org.openmrs.mobile.models.Resource

@Entity(tableName = "visits")
class VisitEntity : Resource() {
    @ColumnInfo(name = "patient_id")
    @SerializedName("patient_id")
    @Expose
    var patientKeyID: Long = 0

    @ColumnInfo(name = "visit_type")
    @SerializedName("visit_type")
    @Expose
    var visitType: String? = null

    @ColumnInfo(name = "visit_place")
    @SerializedName("visit_place")
    @Expose
    var visitPlace: String? = null

    @ColumnInfo(name = "start_date")
    @SerializedName("start_date")
    lateinit var startDate: String

    @ColumnInfo(name = "stop_date")
    @SerializedName("stop_date")
    @Expose
    var stopDate: String? = null

    fun isStartDate(): String {
        return startDate
    }

}
